examples/mongodb-atlas/mongodb-atlas-blueprint.yaml

```diff
@@ -0,0 +1,197 @@
+kind: Blueprint
+metadata:
+  name: mongodb-atlas-blueprint
+actions:
+  backup:
+    outputArtifacts:
+      backupOutput:
+        keyValue:
+          snapshotId: "{{ .Phases.mongoBackup.Output.snapshotId }}"
+    phases:
+    - func: KubeTask
+      name: mongoBackup
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        image: <registry>/<repository>/mongodb-atlas:<tag_name>
+        command:
+          - bash
+          - -o
+          - errexit
+          - -o
+          - pipefail
+          - -c
+          - |
+            cat >/root/.config/atlascli/config.toml <<EOL
+            [default]
+            org_id = '{{ .Object.data.orgid | toString | b64dec }}'
+            output = 'json'
+            private_api_key = '{{ .Object.data.privatekey | toString | b64dec }}'
+            project_id = '{{ .Object.data.projectid | toString | b64dec }}'
+            public_api_key = '{{ .Object.data.publickey | toString | b64dec }}'
+            service = 'cloud'
+            EOL
+            cluster_name='{{ .Object.data.clustername | toString | b64dec }}'
+            atlas backups snapshots create ${cluster_name} --desc "MongoDB Atlas snapshot" -o json > output.json
+            snapshot_id=$(jq -r ".id" output.json)
+            kando output snapshotId ${snapshot_id}
+
+    - func: KubeTask
+      name: WaitForBackupToComplete
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        image: <registry>/<repository>/mongodb-atlas:<tag_name>
+        command:
+          - bash
+          - -o
+          - errexit
+          - -o
+          - pipefail
+          - -c
+          - |
+            cat >/root/.config/atlascli/config.toml <<EOL
+            [default]
+            org_id = '{{ .Object.data.orgid | toString | b64dec }}'
+            output = 'json'
+            private_api_key = '{{ .Object.data.privatekey | toString | b64dec }}'
+            project_id = '{{ .Object.data.projectid | toString | b64dec }}'
+            public_api_key = '{{ .Object.data.publickey | toString | b64dec }}'
+            service = 'cloud'
+            EOL
+            cluster_name='{{ .Object.data.clustername | toString | b64dec }}'
+            isCreated=false
+            isFailed=false
+            timeout=1200
+            while [[ $isCreated != true && $isFailed == false ]];
+            do
+              atlas backups snapshots describe {{ .Phases.mongoBackup.Output.snapshotId }}\
+              --clusterName ${cluster_name} -o json > output.json
+              isCompleted=$(jq -r ".status" output.json)
+              if [ $isCompleted == "failed" ]; then
+                exit 1
+              fi
+              if [ $isCompleted == "completed" ]; then
+                exit 0
+              fi
+              sleep 30
+              if [[ $timeout -le 0 ]]; then
+                echo "Timed out waiting for backup completion."
+                exit 1
+              fi
+              timeout=$((timeout-30))
```

Comment:
Should we use a counter instead of decrementing timeout? It will be easier to understand.

Reply:
fixed